File: exercícios/exemplos/lista9/ex2.scala
object SomaDeVectores { 
  def somarVectores(vetor1: Array[Int], vetor2: Array[Int]): Array[Int] = { 
    require(vetor1.length  ==  vetor2.length  &&  vetor1.length  ==  6,  "Os 
vetores devem ter o mesmo tamanho (6).") 
    vetor1.zip(vetor2).map { case (x, y) => x + y } 
  } 
 
  def main(args: Array[String]): Unit = { 
    val vetor1 = Array(1, 2, 3, 4, 5, 6) 
    val vetor2 = Array(6, 5, 4, 3, 2, 1) 
 
    val resultado = somarVectores(vetor1, vetor2) 
 
    println("Vetor 1: " + vetor1.mkString(", ")) 
    println("Vetor 2: " + vetor2.mkString(", ")) 
    println("Resultado da soma: " + resultado.mkString(", ")) 
  } 
}

File: exercícios/exemplos/lista11/ex2.hs
somaVetores :: [Int] -> [Int] -> [Int] 
somaVetores [] [] = [] 
somaVetores (x:xs) (y:ys) = (x + y) : somaVetores xs ys 
 
main :: IO () 
main = do 
    putStrLn "Digite os elementos do primeiro vetor separados por espaço:" 
    input1 <- getLine 
    let vetor1 = map read $ words input1 :: [Int] 
 
    putStrLn "Digite os elementos do segundo vetor separados por espaço:" 
    input2 <- getLine 
    let vetor2 = map read $ words input2 :: [Int] 
 
    if length vetor1 /= 6 || length vetor2 /= 6 
        then putStrLn "Os vetores devem ter tamanho 6!" 
        else do 
            let resultado = somaVetores vetor1 vetor2 
            putStrLn $ "A soma dos vetores é: " ++ show resultado 
 
 
 
